agent/bench-scripts/pbench-uperf

```diff
@@ -181,280 +182,295 @@ function gen_xml {
-		if [ -n "$1" ]; then
-			log_response_times="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				log_response_times="$1"
+				shift
+			fi
 		;;
 		-p|--protocols)
-		shift
-		if [ -n "$1" ]; then
-			protocols="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				protocols="$1"
+				shift
+			fi
 		;;
 		-C|--client|--clients)
-		shift
-		if [ -n "$1" ]; then
-			clients="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				clients="$1"
+				shift
+			fi
 		;;
 		-S|--server|--servers)
-		shift
-		if [ -n "$1" ]; then
-			servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				servers="$1"
+				shift
+			fi
 		;;
 		--vsock-server|--vsock-servers)
-		shift
-		if [ -n "$1" ]; then
-			vsock_servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				vsock_servers="$1"
+				shift
+			fi
 		;;
 		-c|--config)
-		shift
-		if [ -n "$1" ]; then
-			config="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				config="$1"
+				shift
+			fi
 		;;
 		--start-iteration-num)
-		shift
-		if [ -n "$1" ]; then
-			start_iteration_num=$1
-			shift
-		fi
+			if [ -n "$1" ]; then
+				start_iteration_num=$1
+				shift
+			fi
 		;;
 		--client-node|--client-nodes)
-		shift
-		if [ -n "$1" ]; then
-			client_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				client_nodes="$1"
+				shift
+			fi
 		;;
 		--server-node|--server-nodes)
-		shift
-		if [ -n "$1" ]; then
-			server_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				server_nodes="$1"
+				shift
+			fi
 		;;
 		--sysinfo)
-		shift;
-		if [ -n "$1" ]; then
-			sysinfo="$1"
-			shift;
-		fi
+			if [ -n "$1" ]; then
+				sysinfo="$1"
+				shift
+			fi
 		;;
 		--)
-		shift
-		break
+			break
 		;;
 		*)
-		error_log "[$script_name] bad option, \"$1 $2\""
-		exit 1
-		break
+			error_log "[$script_name] bad option, \"${arg} ${1}\""
+			usage >&2
+			exit 1
 		;;
 	esac
 done
 verify_common_bench_script_options $tool_group $sysinfo
 
-# Verify the number of clients and servers match.
-if [ ! -z "$clients" -a "$postprocess_only" != "y" ]; then
+clients=${clients:-${clients_def}}
+servers=${servers:-${servers_def}}
+
+if [[ "${postprocess_only}" != "y" ]]; then
+	# Boolean value recording if local clients or servers are specified.
+	let check_local=0
+
+	# Verify that the number of clients, servers, and optional vsock servers
+	#  match.
 	let c_cnt=0
-	for client in `echo $clients | sed -e s/,/" "/g`; do
-		let c_cnt=c_cnt+1
+	for client in ${clients//,/ }; do
+		(( c_cnt++ ))
+		if ! is_remote "${client}"; then
+			check_local=1
+		fi
 	done
+	if [[ ${c_cnt} -eq 0 ]]; then
+		error_log "[${script_name}] INTERNAL ERROR - no clients found"
+		exit 1
+	fi
 	let s_cnt=0
-	for server in `echo $servers | sed -e s/,/" "/g`; do
-		let s_cnt=s_cnt+1
+	for server in ${servers//,/ }; do
+		(( s_cnt++ ))
+		if ! is_remote "${server}"; then
+			check_local=1
+		fi
 	done
-	if [ $c_cnt -ne $s_cnt ]; then
+	if [[ ${c_cnt} -eq 0 ]]; then
+		error_log "[${script_name}] INTERNAL ERROR - no servers found"
+		exit 1
+	fi
```

Comment:
While this (third? second?) occurrence is not only "redundant", but wrong ... I assume you wanted `s_cnt` here rather than checking `c_cnt` again ... but it shouldn't be possible either way. 😆

Reply:
Yes, this is an assertion in `bash`.